src/stores/project.store.ts

```diff
@@ -48,10 +52,31 @@ export class ProjectStore {
     return `${instrumentName} ${instrumentCount + 1}`;
   }
 
-  private getSheet(sheetId: SheetId) {
+  public getSheet(sheetId: SheetId) {
     return this.sheetList.find(sheet => sheet.id === sheetId);
   }
```

Comment:
I wonder if it would be better to throw an Error if there's no sheet for a given `sheetId`. TypeScript forces us to deal with the case when we call the `getSheet` store method, since we may potentially receive an undefined return object for an invalid sheetId. 

See https://github.com/supernoteworthy/noteworthy/pull/31/files#diff-8761bd9bcea490857c4a98844b03f444R28

Reply:
Probably agree. On the other hand, returning nulls allows for handling the case where it doesn't exist (without checking it first using another function or try/catch).

I suppose maybe the "trendy" way in lots of programming languages is the trendy Maybe type (or Optional, Option, etc.) I wonder if returning `value | null` is actually pretty close to that in TypeScript (since as you point out handling the null is required).

Feel free to change this, or I am curious to hear your extended thoughts on the benefits/drawbacks of Error vs. null, since I should probably develop a real opinion!!

Reply:
Hmmm it's my loosely held opinion that it is dependent on the situation and what your expectations are for that function call.  For example, I assume that we are never going to provide an invalid `sheetId` to  `getSheet()` function and if we did provide an invalid `sheetId` argument we would want it to throw an error immediately without having to check that we are getting a `null` sheetObject back. This would also means we have less conditional logic each time we call the function.

But throwing an error immediately limits our flexibility in how we can deal with the result of the `getSheet()` function. I assume this would only be a problem if we were expecting to make calls to function where there would be cases where we would try to get a sheet to do something but not actually do anything if `null` is returned.

I was thinking about using a Maybe type as well, but the Maybe type still requires you to check if the type is null or not later.  I don't think TypeScript has a native Maybe type, but this did work for me: `type Maybe<t> = t | null` 

Just my 2 cents! Probably not worth a whole new PR, but glad to be thinking about code style. Open to any suggestions or thoughts as well!